src/Enrich.hs

```diff
@@ -83,18 +83,16 @@ mkEthEnvBounds vars = catMaybes $ mkBound <$> nub vars
       Nonce -> AbiUIntType 256
 
 -- | extracts bounds from the AbiTypes of Integer values in storage
-mkStorageBounds :: Store -> [Either StorageLocation StorageUpdate] -> [Exp Bool]
-mkStorageBounds store refs
-  = catMaybes $ mkBound <$> refs
+mkStorageBounds :: Store -> [Rewrite] -> [Exp Bool Untimed]
```

Comment:
Because we're enriching a typed but unrefined AST. So the `Rewrite` that we take in will be untimed, and so will the `TStorageItem` which `fromItem` receives, so it can't produce an expression with arbitrary timing, and up the calling chain it goes. But maybe this is wrong? The only reason why I chose to enrich unrefined ASTs was because then I wouldn't have to change (or understand) the code in `Enrich`.

Reply:
Additionally, we actually need an untimed expression in the end, because we're inserting the storage bound into a `Syntax.Typed.Invariant`, which is defined as `Syntax.TimeAgnostic.Invariant Untimed`. So it will require that the storage bounds are untimed. But again, maybe we should enrich refined expressions instead.

Reply:
I'm not sure that I have a super strong opinion on the ordering of enrich and refine. I guess it feels slightly more natual to me to enrich after refining but I can't think of any important reason why, so it's just an aesthetic thing. Either way I think we can leave that for antother PR if we decide to make that change.